src/sentry/rules/actions/utils.py

```diff
@@ -31,41 +31,38 @@ def check_value_changed(
 
 
 def generate_diff_labels(
-    present_state: dict[str, Any],
-    prior_state: dict[str, Any],
+    present_state: list[dict[str, Any]],
+    prior_state: list[dict[str, Any]],
     rule: Rule,
-    changed_data: DefaultDict[str, list[str]],
     key: str,
 ) -> DefaultDict[str, list[str]]:
     added_statement = "Added {} '{}'"
     removed_statement = "Removed {} '{}'"
     added_items = []
-
+    changed_data: DefaultDict[str, list[str]] = defaultdict(list)
     # Added items that are not seen in the prior rule data
-    for data in prior_state.get(key, []):
-        if data not in present_state.get(key, []):
+    for data in prior_state:
+        if data not in present_state:
             label = generate_rule_label(rule.project, rule, data)
-            changed_data[data["id"]].append(added_statement.format(key[:-1], label))
+            changed_data[data["id"]].append(added_statement.format(key, label))
             added_items.append(data["id"])
 
     # Check if the id is in both but the data has changed
     for item in added_items:
         if changed_data.get(item):
```

Comment:
So - we're populating both `added_items` and `changed_data` in the same precompile above...

When will `changed_data.get(item)` ever be false?